drt_1d.py

```diff
@@ -88,10 +195,50 @@ def construct_transition_matrix(num_voxels:int, voxel_size:float, sigma_s:np.nda
     for i in range(num_voxels):
         for j in range(num_voxels):
             if i == j:
-                # Set diagonal values to one based on defintion of streaming operator
-                K_trans[i, j] = 1
+                # Diagonal element (same voxel)
+                # Retrieve the list of positions for the current voxel i (same as j in this case)
+                positions_in_voxel = positions.get(str(i))
+                
+                # Check if there are multiple positions within this voxel
+                if len(positions_in_voxel) > 1:
+                    # Calculate the total number of position pairs (combinations of 2)
+                    position_pairs = itertools.combinations(positions_in_voxel, 2)
+                    
+                    streaming_operator_sum = 0.0
+                    num_position_pairs = 0
+                    for position_a, position_b in position_pairs:
+                        # Calculate the streaming operator for the pair (both directions)
+                        streaming_operator_sum += calculate_K_trans(i, j, position_a, position_b, voxel_size, sigma_s)
+                        streaming_operator_sum += calculate_K_trans(i, j, position_b, position_a, voxel_size, sigma_s)
+                        num_position_pairs += 2
+                    
+                    # Compute average streaming operator for this voxel and assign it to the transition matrix
+                    K_trans[i, j] = streaming_operator_sum / num_position_pairs
+                    
+                    logging.debug("Voxel %d has %d positions, average streaming operator: %f", i, len(positions_in_voxel), K_trans[i, j])
+                else:
+                    # If there is only one position, set diagonal element to 1
+                    K_trans[i, j] = 1
+                    logging.debug("Voxel %d has a single position, set K_trans[%d, %d] to 1", i, i, j)
             else:
-                K_trans[i, j] = calculate_K_trans(i, j, voxel_size, sigma_s)
+                # Off-diagonal element (different voxels)
+                positions_in_voxel_i = positions.get(str(i))
+                positions_in_voxel_j = positions.get(str(j))
```

Comment:
Why not:
```suggestion
                positions_in_voxel_i = positions[str(i)]
                positions_in_voxel_j = positions[str(j)]
```

(even better if you convert the keys to integers when you read the data)